File: src/main/java/com/team3/gudit/auth/exception/AuthErrorCodeProvider.java
package com.team3.gudit.auth.exception;

import com.team3.gudit.global.exception.ErrorCode;
import com.team3.gudit.global.exception.ErrorCodeProvider;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class AuthErrorCodeProvider implements ErrorCodeProvider<AuthErrorCode> {
    @Override
    public String getDomain() {
        return "AUTH";
    }

    @Override
    public List<ErrorCode> getErrorCodes() {
        return List.of(AuthErrorCode.values());
    }
}

File: src/main/kotlin/com/team3/gudit/auth/exception/AuthErrorCodeProvider.kt
package com.team3.gudit.auth.exception

import com.team3.gudit.global.exception.ErrorCode
import com.team3.gudit.global.exception.ErrorCodeProvider
import org.springframework.stereotype.Component

@Component
class AuthErrorCodeProvider : ErrorCodeProvider<AuthErrorCode> {
    override fun getDomain(): String = "AUTH"
    override fun getErrorCodes(): List<ErrorCode> = AuthErrorCode.entries.toList()
}
